src/core/server/saved_objects/migrations/model/model.ts

```diff
@@ -72,7 +72,26 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   if (stateP.controlState === 'INIT') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
 
-    if (Either.isRight(res)) {
+    if (Either.isLeft(res)) {
+      const left = res.left;
+      if (isLeftTypeof(left, 'cluster_routing_allocation_disabled')) {
+        return {
+          ...stateP,
+          controlState: 'FATAL',
+          reason: `The elasticsearch cluster has cluster routing allocation disabled. To proceed, please remove the cluster routing allocation settings with PUT /_cluster/settings {"transient": {"cluster.routing.allocation.enable": null}, "persistent": {"cluster.routing.allocation.enable": null}}`,
```

Comment:
Gives the full command on how to resolve the cause of the failure. There's a [follow-up issue](https://github.com/elastic/kibana/issues/126864) to add a link to the docs